apps/mbdb/fields.py

```diff
@@ -25,76 +25,79 @@ class PickledObject(str):
 
 
 class PickledObjectField(models.Field):
-    __metaclass__ = models.SubfieldBase
 
     def __init__(self, *args, **kwargs):
         super(PickledObjectField, self).__init__(*args, **kwargs)
-        # By default, South will make this field an index. MySQL doesn't
-        # accept TEXT fields to be indexed, and will raise an error.
+        # MySQL doesn't accept TEXT fields to be indexed,
+        # and will raise an error.
         # We thus need to force this to *not* be indexed by MySQL.
         self.db_index = False
 
+    def from_db_value(self, value, expression, connection, context):
+        if value is None:
+            return value
+        return pickle.loads(str(value))
+
     def to_python(self, value):
         if value is None:
             return value
-        if isinstance(value, PickledObject):
-            # If the value is a definite pickle; and an error is raised in
-            # de-pickling it should be allowed to propogate.
+        try:
             return pickle.loads(str(value))
-        else:
-            try:
-                return pickle.loads(str(value))
-            except:
-                # If an error was raised, just return the plain value
-                return value
-
-    def get_db_prep_save(self, value, connection):
-        if value is not None and not isinstance(value, PickledObject):
-            if isinstance(value, str):
-                # normalize all strings to unicode, like django does
-                value = unicode(value)
-            value = PickledObject(pickle.dumps(value))
+        except:
```

Comment:
I looked up https://docs.python.org/2/library/pickle.html#pickle.UnpicklingError, and it's quite liberal in which errors you shall expect :-/